remote/remote_state_s3.go

```diff
@@ -387,21 +389,17 @@ func checkIfVersioningEnabled(s3Client *s3.S3, config *RemoteStateConfigS3, terr
 
 // Create the given S3 bucket and enable versioning for it
 func CreateS3BucketWithVersioningSSEncryptionAndAccessLogging(s3Client *s3.S3, config *ExtendedRemoteStateConfigS3, terragruntOptions *options.TerragruntOptions) error {
-	err := CreateS3Bucket(s3Client, &config.remoteStateConfigS3, terragruntOptions)
+	err := CreateS3Bucket(s3Client, aws.String(config.remoteStateConfigS3.Bucket), terragruntOptions)
 
 	if err != nil {
-		if isBucketAlreadyOwnedByYourError(err) {
-			terragruntOptions.Logger.Printf("Looks like someone is creating bucket %s at the same time. Will not attempt to create it again.", config.remoteStateConfigS3.Bucket)
+		if isBucketAlreadyOwnedByYouError(err) {
+			terragruntOptions.Logger.Printf("Looks like you're already creating bucket %s at the same time. Will not attempt to create it again.", config.remoteStateConfigS3.Bucket)
 			return WaitUntilS3BucketExists(s3Client, &config.remoteStateConfigS3, terragruntOptions)
 		}
 
 		return err
 	}
 
-	if err := WaitUntilS3BucketExists(s3Client, &config.remoteStateConfigS3, terragruntOptions); err != nil {
```

Comment:
Apologies if I missed the discussion around this, but why remove the wait?

Reply:
Good spot, Jim! I meant to comment on this, and confirm this is a duplicated piece of logic. Apologies for forgetting to do so earlier. 

I've removed it thinking it was a duplicate. See commit https://github.com/gruntwork-io/terragrunt/pull/1400/commits/e77cde2ecef9eb2f576c6c6e825d7d10edafb157

Reply:
Why do you think it's duplicated?

Reply:
_Actually, after re-reading it for the 5th time, I think I was massively misreading what the flow of the logic is 🤦 ._

_What I was thinking:_ It's already handling the error & waiting for the bucket to exist.= in the guard block (line 395-401)

_What's I think is actually happening. P;ease correct me if I'm wrong in this :) _ It's only surpressing the 'OwnedByYouError' so it doesn't exit prematurely & waits instead. But if there's no error thrownm we still want to make sure the bucket has been created fine - hence the second `WaitUntilS3...`

Reply:
Yea, the check on 395-401 is an extra check for concurrent bucket creation: e.g., if you run `terragrunt apply-all`, it will execute on a bunch of modules in parallel, all of which might try to create the exact same bucket at the same time.

The check `WaitUntilS3BucketExists` I highlighted in this comment ensures that _after_ the bucket has been created, we wait long enough that we can "see" it in the AWS APIs. This is to work around the fact that AWS APIs are async and eventually consistent. Without this check, the code would move to the following parts, which try to modify the S3 bucket, and potentially hit errors because the bucket information hasn't propagated yet.

Reply:
you're absolutely right - thanks for spotting & explaining 🎈 
*Reverting this commit now*